src/layman/layer/filesystem/gdal_test.py

```diff
@@ -0,0 +1,31 @@
+from test import util as test_util
+import pytest
+from layman import LaymanError
+from . import gdal
+
+
+@pytest.mark.parametrize('file_path, exp_error', [
+    ('sample/layman.layer/sample_jp2_rgb.jp2', None),
+    ('sample/layman.layer/sample_tif_rgb.tif', None),
+    ('sample/layman.layer/sample_tiff_rgba.tiff', None),
+    ('sample/layman.layer/sample_tif_tfw_rgba.tif', None),
+    ('sample/layman.layer/sample_tif_colortable_nodata.tif', None),
+    ('sample/layman.layer/sample_tif_grayscale_nodata.tif', None),
+    ('sample/layman.layer/sample_tif_rg.tif', {'http_code': 400,
+                                               'code': 2,
+                                               'detail': {'parameter': 'file',
+                                                          'expected': 'Any of color interpretations [Gray], '
+                                                                      '[Gray, Alpha], [Palette], [Red, Green, Blue], '
+                                                                      '[Red, Green, Blue, Alpha].',
+                                                          'found': ['Red', 'Green']
+                                                          },
+                                               }),
+])
+@pytest.mark.usefixtures('ensure_layman')
+def test_assert_valid_raster(file_path, exp_error):
+    if exp_error is None:
+        gdal.assert_valid_raster(file_path)
+    else:
+        with pytest.raises(LaymanError) as exc_info:
+            gdal.assert_valid_raster(file_path)
+        test_util.assert_error(exp_error, exc_info)
```

Comment:
See https://stackoverflow.com/questions/20274987/how-to-use-pytest-to-check-that-error-is-not-raised